Update to CF v4

jimmynotjim opened the pull request:
Updated the project to use the latest version of Capital Framework.

## Additions

- Added new color variables from Capital Framework

## Removals

- Removed duplicate styles that have already been ported from cf-enhancements

## Changes

- Updated the CF modules to version 4.1.0 in package.json.
- Updated all the elements to use the correct class names.

## Testing

- run `grunt less && grunt scripts`.
- run `npm start` and navigate through the site.

## Review

- @Scotchester  
- @cfarm 
- @contolini 

[Preview this PR without the whitespace changes](?w=0)

## Screenshots


## Notes

- There are a few color variable that aren't using colors from our palette. I'm leaving them as is for now because they should really be handled in the cfpb-theme.

## Todos

- Remove any duplicate styles between this project and Capital Framework.
- Update the project to use the cfpb-theme module.

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

jimmynotjim committed:
Updated the project to cfv4

This updates the entire project to use the latest release of Capital Framework v4, currently v4.3.0. Includes changes to markup patterns and less variables, mixins, and atomic organization.

Scotchester commented:
This is not a problem with your PR, but looking at the expandables page, I realize that our `:last-child { margin-bottom: 0; }` rule has eliminated the bottom spacing in all expandables, which were counting on their last child's bottom margin to be there. I'll file an issue in capital-framework.

Scotchester commented:
Got a couple issues on the Tables page:

<img width="871" alt="screen shot 2017-03-23 at 15 50 02" src="https://cloud.githubusercontent.com/assets/1044670/24267380/7b19928c-0fe0-11e7-9be2-e026f9589cd8.png">

<img width="897" alt="screen shot 2017-03-23 at 15 50 22" src="https://cloud.githubusercontent.com/assets/1044670/24267394/8199390a-0fe0-11e7-8e66-8a6aec6b3c27.png">

jimmynotjim commented:
Updated with @Scotchester fix for pagination and markup corrections for the tables. I removed the 100% inline widths because I'd like to solve that issue in cf-tables. I don't know if we should hold this up for that or not though.

jimmynotjim commented:
It's going to get messy dealing with ongoing changes to the less files with this project on v3. Can we merge this?

Scotchester commented:
Testing this again locally.

> I removed the 100% inline widths because I'd like to solve that issue in cf-tables.

Are you proposing we set all table widths to 100% by default? I'm not sure we want to do that.

But anyway, a more efficient way to do it in this project would be to add a style for it in the project's stylesheets. I'll commit that, along with a fix for the pagination style to correctly show how it would look on a small screen.

We should also decide whether the expandable bottom padding issue should be a blocker to moving forward with this. As soon as we publish, those examples will look broken.

jimmynotjim commented:
@Scotchester agreed that the best option would be setting them in the main stylesheet of this project. As far as the expandables, yes, we should fix that first. I can do that after I make the hero updates today.

Scotchester commented:
Whoops, pushed that commit to my fork yesterday afternoon. Pushed it here now.

jimmynotjim commented on Gruntfile.js:
```diff
@@ -312,7 +312,7 @@ module.exports = function(grunt) {
       },
       content: {
-        files: ['**/*.html', '/downloads/*.md', '/guides/*.md', '/identity/*.md', '/ui-toolkit/*.md'],
+        files: ['index.html', '**/*.md'],
```
Why are we even converting these? Let's let Jekyll handle the markdown files, just like we did on the CF docs site.

Scotchester replied:
They're not being converted, just triggering the copy task. The main use case would be adding new images to a page. On saving the MD file, images will be recopied from `./src/static/img/` (everything in `src` is ignored by Jekyll) to `./static/img`, where they'll be picked up by Jekyll's watcher.

We could consider altering/getting rid of the `src`/"`dist`" model that's in use here and just keep static files that need no building in `./static`.

jimmynotjim commented on src/static/css/content.less:
```diff
@@ -765,3 +765,32 @@ ul {
+    width: 100%;
+}
+
+// Display pagination examples in half columns as their mobile view
```
I still need to try the iframe suggestion @niqjohnson had. Maybe I can find time on Friday.

jimmynotjim commented:
Expandables bug is being addressed at https://github.com/cfpb/AtomicComponent/pull/4 and https://github.com/cfpb/capital-framework/pull/471

jimmynotjim commented:
@Scotchester I think I've addressed everything we discovered in this process. Hopefully we can merge this and keep moving on other improvements.

Scotchester commented on Gemfile.lock:
```diff
@@ -0,0 +1,200 @@
+GEM
```
How about we go ahead and remove & ignore `Gemfile.lock`?

jimmynotjim replied:
That will go away when I rebase, @marteki removed that already.

Scotchester commented on package.json:
```diff
@@ -20,16 +20,16 @@
+    "cf-layout": "^4.1.0",
+    "cf-pagination": "^4.1.0",
+    "cf-tables": "^4.1.0",
+    "cf-typography": "^4.1.0",
```
Several of these need to get updated to their latest releases. Easiest thing to reference is the badges at the top of the capital-framework README.

jimmynotjim replied:
@Scotchester the `^` takes care of that for you, run `npm update`.

Scotchester reviewed (changes requested):
Need a quick update to the `package.json` file to make sure I'm testing things totally cleanly, rather than relying on npm links.

Scotchester commented:
:+1:
